// Copyright 2022 The Cobalt Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package on_device_tests_gateway;

// Interface exported by the server.
service on_device_tests_gateway {
  // A dumb proxy RPC service that passes user defined command line options
  // to the on-device tests gateway and streams back output in real time.
  rpc exec_command(OnDeviceTestsCommand)
      returns (stream OnDeviceTestsResponse) {}

  // A dumb proxy RPC service that passes user defined command line options
  // to the on-device tests gateway and streams back output in real time.
  rpc exec_watch_command(OnDeviceTestsWatchCommand)
      returns (stream OnDeviceTestsResponse) {}
}

// Working directory and command line arguments to be passed to the gateway.
message OnDeviceTestsCommand {
  string token = 2;
  repeated string labels = 8;
  repeated TestRequest test_requests = 24;
}

message TestRequest {
  // Args picked up by MH, e.g. "name1=value", "name2=value.
  repeated string test_args = 1;
  // Args sent to test binary/apk, e.g. "--arg1", "--arg2=value".
  repeated string test_cmd_args = 2;
  // Files to send to device, e.g. "build_apk=/bigstore/bucket/test.apk",
  // "test_apk=/bigstore/bucket/test.apk",
  repeated string files = 3;
  // e.g. "gcs_result_path=gs://some/gcs/path"
  repeated string params = 4;
  // "sabrina" or "boreal"
  string device_type = 5;
  // "shared" or "maneki"
  string device_pool = 6;
}

// Working directory and command line arguments to be passed to the gateway.
message OnDeviceTestsWatchCommand {
  // Next ID: 6
  string token = 2;
  string session_id = 3;
  bool dry_run = 5;
}

// Response from the on-device tests.
message OnDeviceTestsResponse {
  // Next ID: 2
  string response = 1;
}
